src/ssl.c

```diff
@@ -19376,20 +19376,11 @@ static void *wolfSSL_d2i_X509_fp_ex(XFILE file, void **x509, int type)
     goto _exit;
 
 err_exit:
-    if(newx509 != NULL){
-        if(type == CERT_TYPE)
-            wolfSSL_X509_free((WOLFSSL_X509*)newx509);
-    #ifdef HAVE_CRL
-        else if(type == CRL_TYPE) {
-            wolfSSL_X509_CRL_free((WOLFSSL_X509_CRL*)newx509);
-        }
-    #endif
```

Comment:
This change will cause resources to go unfreed in the cases of `type != PKCS12_TYPE`

Reply:
According to what I see in the code, if `type == CERT_TYPE` or `type == CRL_TYPE`, the program will go to err_exit only if `newx509` is NULL, so these conditions are pointless.

Reply:
Right, I see it now. Thanks for walking me through it.